documentation/development-docs/azure-powershell-design-guidelines.md

```diff
@@ -0,0 +1,308 @@
+
+#### Pascal Case
+
+Similar to cmdlets (mentioned above), parameters should follow pascal casing.From the [Strongly Encouraged Development Guidelines](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx):
+
+> _Use Pascal case for cmdlet names. In other words, capitalize the first letter of the verb and all terms used in the noun. For example, "Clear-ItemProperty"._
+
+#### Singularity
+
+From the [Strongly Encouraged Development Guidelines](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx):
+
+> _Avoid using plural names for parameters whose value is a single element. This includes parameters that take arrays or lists because the user might supply an array or list with only one element._
+>
+>_Plural parameter names should be used only in those cases where the value of the parameter is always a multiple-element value. In these cases, the cmdlet should verify that multiple elements are supplied, and the cmdlet should display a warning to the user if multiple elements are not supplied._
+
+#### Parameter Alias
+
+If you there is a separate nomenclature for the parameter name, or if you would like to shorten the name of the parameter so it's easier to remember, you can add an alias attribute to your parameter to allow for this functionality.
+
+### Bool vs. SwitchParameter
+
+Parameters of type `bool` are _strongly_ discouraged in PowerShell. The `SwitchParameter` type of a parameter acts a flag that signals whether or not some action should be taken based on if the parameter was provided or not. From the [Strongly Encouraged Development Guidelines](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx):
+
+> _If your parameter takes only true and false, define the parameter as type SwitchParameter. A switch parameter is treated as true when it is specified in a command. If the parameter is not included in a command, Windows PowerShell considers the value of the parameter to be false. Do not define Boolean parameters._
+
+### Closed Set of Values
+
+If there is a closed set of values applicable for a given parameter, use either a `ValidateSet` or an enumeration type. This functionality allows users to tab through the different values they can provide. From the [Strongly Encouraged Development Guidelines](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx):
+
+> _There are two ways to create a parameter whose value can be selected from a set of options._
+> 
+> - _Define an enumeration type (or use an existing type) that specifies the valid values. Then, use the enumeration type to create a parameter of that type._
+> - _Add the **ValidateSet** attribute to the parameter declaration._
+
```

Comment:
Maybe explain Mandatory and Position? Unless you think that is self-explanatory.

Reply:
@maddieclayton I think these properties are self-explanatory, but I will create a follow-up document that explains the attributes and properties found in PowerShell from `System.Management.Automation`, similar to the examples we have [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines)